tests/test_second_kick.py

```diff
@@ -0,0 +1,271 @@
+                                screen_size=102.4, screen_scale=0.05)
+        atm.instantiate(kmin=kcrit)
+        psf = galsim.PhaseScreenPSF(atm, lam=500, t0=0, exptime=10, aper=aper, time_step=0.1)
+        phaseImg = psf.drawImage(nx=64, ny=64, scale=0.02)
+        sk = galsim.SecondKick(lam=500, r0=r0, diam=diam, obscuration=obscuration, L0=L0,
+                               kcrit=kcrit)
+        skImg = sk.drawImage(nx=64, ny=64, scale=0.02)
+        phaseMom = galsim.hsm.FindAdaptiveMom(phaseImg)
+        skMom = galsim.hsm.FindAdaptiveMom(skImg)
+
+        print(phaseMom.moments_sigma, skMom.moments_sigma)
+        np.testing.assert_allclose(phaseMom.moments_sigma, skMom.moments_sigma, rtol=2e-2)
+
+        # fig, axes = plt.subplots(nrows=1, ncols=2)
+        # vmin = -6
+        # vmax = -1
+        # phim = axes[0].imshow(np.log10(phaseImg.array), vmin=vmin, vmax=vmax)
+        # axes[0].set_title("PhasePSF")
+        # skim = axes[1].imshow(np.log10(skImg.array), vmin=vmin, vmax=vmax)
+        # axes[1].set_title("SecondKick")
+        # fig.tight_layout()
+        # plt.show()
+
+@timer
+def test_sk_scale():
+    """Test sk scale argument"""
+    kwargs = {'lam':500, 'r0':0.2, 'L0':25.0, 'diam':4.0, 'flux':2.2, 'obscuration':0.3}
+    sk_arcsec = galsim.SecondKick(scale_unit=galsim.arcsec, **kwargs)
+    sk_arcmin = galsim.SecondKick(scale_unit='arcmin', **kwargs)
+    do_pickle(sk_arcmin)
+
+    np.testing.assert_almost_equal(sk_arcsec.flux, sk_arcmin.flux)
+    np.testing.assert_almost_equal(sk_arcsec.kValue(0.0, 0.0), sk_arcmin.kValue(0.0, 0.0))
+    np.testing.assert_almost_equal(sk_arcsec.kValue(0.0, 10.0), sk_arcmin.kValue(0.0, 600.0))
+    np.testing.assert_almost_equal(sk_arcsec.xValue(0.0, 6.0), sk_arcmin.xValue(0.0, 0.1))
+
+    img1 = sk_arcsec.drawImage(nx=32, ny=32, scale=0.2)
+    img2 = sk_arcmin.drawImage(nx=32, ny=32, scale=0.2/60.0)
+    np.testing.assert_almost_equal(img1.array, img2.array)
+
+
+@timer
+def test_sk_ne():
+    gsp = galsim.GSParams(maxk_threshold=1.1e-3, folding_threshold=5.1e-3)
+
+    objs = [galsim.SecondKick(lam=500.0, r0=0.2, diam=4.0),
+            galsim.SecondKick(lam=550.0, r0=0.2, diam=4.0),
+            galsim.SecondKick(lam=500.0, r0=0.25, diam=4.0),
+            galsim.SecondKick(lam=500.0, r0=0.25, diam=4.2),
+            galsim.SecondKick(lam=500.0, r0=0.25, diam=4.2, obscuration=0.4),
+            galsim.SecondKick(lam=500.0, r0=0.25, diam=4.2, L0=1e11),
+            galsim.SecondKick(lam=500.0, r0=0.25, diam=4.2, kcrit=1.234),
+            galsim.SecondKick(lam=500.0, r0=0.25, diam=4.2, flux=2.2),
+            galsim.SecondKick(lam=500.0, r0=0.25, diam=4.2, scale_unit='arcmin'),
+            galsim.SecondKick(lam=500.0, r0=0.2, diam=4.0, gsparams=gsp)]
+    all_obj_diff(objs)
+
+
+if __name__ == '__main__':
+    from argparse import ArgumentParser
+    parser = ArgumentParser()
+    parser.add_argument("--slow", action='store_true', help="Run slow tests")
```

Comment:
Again, I'm going to push toward having the default be to include at least more, if not all, of the "slow" test.  Timings on my laptop:

slow = false: 0:34
slow = true: 8:09
nosetests (so no name==main stuff): 0:17

So the non-main version is probably a little slower than we want.  And already more of the slow tests could be moved into the default main runs.  I'd like to get all of them in by default, but 8 min is probably too long for that.

I suspect some of this will speed up when you fix the xValue code that I think is still the slow version.  I'll probably profile this to see if I can speed things up the way I did for VK.  But I'll wait to do that until you've addressed the current round of comments.  If you feel like trying your hand at profiling and optimizing, feel free.  But if not, I don't mind doing that.  If we can get the slow run down to a couple minutes, then I'd like to make it the default to include them all.